compiler/rustc_const_eval/src/lib.rs

```diff
@@ -51,7 +51,107 @@ pub fn provide(providers: &mut Providers) {
     };
 }
 
+use rustc_middle::mir::{
+    BasicBlockData, ConstOperand, NullOp, Operand, Rvalue, StatementKind, SwitchTargets,
+    TerminatorKind,
+};
+use rustc_middle::ty::{Instance, TyCtxt};
+
 /// `rustc_driver::main` installs a handler that will set this to `true` if
 /// the compiler has been sent a request to shut down, such as by a Ctrl-C.
 /// This static lives here because it is only read by the interpreter.
 pub static CTRL_C_RECEIVED: AtomicBool = AtomicBool::new(false);
+
+/// If this basic block ends with a [`TerminatorKind::SwitchInt`] for which we can evaluate the
+/// dimscriminant in monomorphization, we return the discriminant bits and the
+/// [`SwitchTargets`], just so the caller doesn't also have to match on the terminator.
+pub fn try_const_mono_switchint<'a, 'tcx>(
```

Comment:
Please find a better place for this function -- the crate root is not a good location for any code.

Worst-case, put it in `util`.